### ISSUE TITLE
Error in JPX calendar: 2016-01-04 was a trading day

### ISSUE THREAD

**Issue by vss888** [rsheftel/pandas_market_calendars#35]
```
import pandas_market_calendars as mcal
jpx = mcal.get_calendar('JPX')
jpx_s = jpx.schedule(start_date='2016-01-01', end_date='2016-01-31')
print(jpx_s)
```

>                          market_open              market_close
> 2016-01-05 2016-01-05 00:00:00+00:00 2016-01-05 06:00:00+00:00
> 2016-01-06 2016-01-06 00:00:00+00:00 2016-01-06 06:00:00+00:00
> 2016-01-07 2016-01-07 00:00:00+00:00 2016-01-07 06:00:00+00:00
> 2016-01-08 2016-01-08 00:00:00+00:00 2016-01-08 06:00:00+00:00
> 2016-01-12 2016-01-12 00:00:00+00:00 2016-01-12 06:00:00+00:00
> 2016-01-13 2016-01-13 00:00:00+00:00 2016-01-13 06:00:00+00:00
> 2016-01-14 2016-01-14 00:00:00+00:00 2016-01-14 06:00:00+00:00
> 2016-01-15 2016-01-15 00:00:00+00:00 2016-01-15 06:00:00+00:00
> 2016-01-18 2016-01-18 00:00:00+00:00 2016-01-18 06:00:00+00:00
> 2016-01-19 2016-01-19 00:00:00+00:00 2016-01-19 06:00:00+00:00
> 2016-01-20 2016-01-20 00:00:00+00:00 2016-01-20 06:00:00+00:00
> 2016-01-21 2016-01-21 00:00:00+00:00 2016-01-21 06:00:00+00:00
> 2016-01-22 2016-01-22 00:00:00+00:00 2016-01-22 06:00:00+00:00
> 2016-01-25 2016-01-25 00:00:00+00:00 2016-01-25 06:00:00+00:00
> 2016-01-26 2016-01-26 00:00:00+00:00 2016-01-26 06:00:00+00:00
> 2016-01-27 2016-01-27 00:00:00+00:00 2016-01-27 06:00:00+00:00
> 2016-01-28 2016-01-28 00:00:00+00:00 2016-01-28 06:00:00+00:00
> 2016-01-29 2016-01-29 00:00:00+00:00 2016-01-29 06:00:00+00:00

2016-01-04 is not in the table, however,  was a trading day:

http://markets.on.nytimes.com/research/markets/holidays/holidays.asp?display=market&timeOffset=-2&exchange=TYO

**Comment by vss888** [comment]
Another bug: 2015-08-11 is marked as holiday, but it was not:

`print( jpx.schedule(start_date='2015-08-01', end_date='2015-08-31') )`

```

>                          market_open              market_close
> 2015-08-03 2015-08-03 00:00:00+00:00 2015-08-03 06:00:00+00:00
> 2015-08-04 2015-08-04 00:00:00+00:00 2015-08-04 06:00:00+00:00
> 2015-08-05 2015-08-05 00:00:00+00:00 2015-08-05 06:00:00+00:00
> 2015-08-06 2015-08-06 00:00:00+00:00 2015-08-06 06:00:00+00:00
> 2015-08-07 2015-08-07 00:00:00+00:00 2015-08-07 06:00:00+00:00
> 2015-08-10 2015-08-10 00:00:00+00:00 2015-08-10 06:00:00+00:00
> 2015-08-12 2015-08-12 00:00:00+00:00 2015-08-12 06:00:00+00:00
> 2015-08-13 2015-08-13 00:00:00+00:00 2015-08-13 06:00:00+00:00
> 2015-08-14 2015-08-14 00:00:00+00:00 2015-08-14 06:00:00+00:00
> 2015-08-17 2015-08-17 00:00:00+00:00 2015-08-17 06:00:00+00:00
> 2015-08-18 2015-08-18 00:00:00+00:00 2015-08-18 06:00:00+00:00
> 2015-08-19 2015-08-19 00:00:00+00:00 2015-08-19 06:00:00+00:00
> 2015-08-20 2015-08-20 00:00:00+00:00 2015-08-20 06:00:00+00:00
> 2015-08-21 2015-08-21 00:00:00+00:00 2015-08-21 06:00:00+00:00
> 2015-08-24 2015-08-24 00:00:00+00:00 2015-08-24 06:00:00+00:00
> 2015-08-25 2015-08-25 00:00:00+00:00 2015-08-25 06:00:00+00:00
> 2015-08-26 2015-08-26 00:00:00+00:00 2015-08-26 06:00:00+00:00
> 2015-08-27 2015-08-27 00:00:00+00:00 2015-08-27 06:00:00+00:00
> 2015-08-28 2015-08-28 00:00:00+00:00 2015-08-28 06:00:00+00:00
> 2015-08-31 2015-08-31 00:00:00+00:00 2015-08-31 06:00:00+00:00
> 
```
http://markets.on.nytimes.com/research/markets/holidays/holidays.asp?display=market&timeOffset=-3&exchange=TYO

**Comment by vss888** [comment]
Another bug: 2015-05-06 was a holiday (Constitution Day), but it is marked as a trading day in the pandas_market_calendars.

print( jpx.schedule(start_date='2015-05-01', end_date='2015-05-31') )

http://markets.on.nytimes.com/research/markets/holidays/holidays.asp?display=market&timeOffset=-3&exchange=TYO

**Comment by vss888** [comment]
Another bug: 2015-09-22 was a holiday (Bridge Holiday), but it is marked as a trading day in the pandas_market_calendars.

print( jpx.schedule(start_date='2015-09-01', end_date='2015-095-30') )

http://markets.on.nytimes.com/research/markets/holidays/holidays.asp?display=market&timeOffset=-3&exchange=TYO

**Comment by rsheftel** [comment]
@gabalese these are JPX calendar issues/questions, do you have an opinion?

**Comment by vss888** [comment]
Guys, in fact since the beginning of 2010 to the end of 2018 you have 33(!) errors in the JPX holiday calendar. The logic of calculating the holidays does not seem to be reliable: it is probably better to just list them in the code.

**Comment by vss888** [comment]
@rsheftel, I would just use the NY Times web-site to get the list of dates. By changing the link

http://markets.on.nytimes.com/research/markets/holidays/holidays.asp?display=market&timeOffset=-30&exchange=TYO     (note, the **timeOffset=-30** part)

one could go up to 30 years back and up 30 years in the future (1988 to 2048). Which means that if it is done correctly now, we will likely not have to worry about it for the next 30 years and will have at least 30 years of correct historical holidays. It is probably possible even to do it algorithmically via web-scraping in Python. I have just done a simple test: I timed how much time it takes to create a list of holidays manually for one year (the result is below) and it took exactly one minute - if I did it for all years at once, it would be even faster. That means that to create manually a correct list of holidays for JPX for 61 years would take less than 1 hour:

01/01/2048 New Year's Day
01/02/2048 Bank Holiday 2
01/03/2048 Bank Holiday 3
01/13/2048 Coming of Age (Adults') Day
02/11/2048 National Founding Day
03/20/2048 Vernal Equinox
04/29/2048 Showa Day (formerly Greenery Day)
05/04/2048 Greenery Day (formerly National Holiday)
05/05/2048 Children's Day
05/06/2048 Constitution Day OBS
07/20/2048 Marine Day
08/11/2048 Mountain Day
09/21/2048 Respect for the Aged Day
09/22/2048 Autumn Equinox
10/12/2048 Health-Sports Day
11/03/2048 Culture Day
11/23/2048 Labour Thanksgiving Day
12/23/2048 Emperor's Birthday
12/31/2048 New Year's Eve
